blockchain/txvalidator.go

```diff
@@ -1271,6 +1284,157 @@ func (b *BlockChain) checkRegisterCRTransaction(txn *Transaction) error {
 	return nil
 }
 
+func (b *BlockChain) getDIDAddress(info *payload.CRInfo) (string, error) {
```

Comment:
nice extraction and you can remove the same logic from checkRegisterCRTransaction method

Reply:
Ok, already removed, my mistake.

Reply:
see 2ab823fe17d72